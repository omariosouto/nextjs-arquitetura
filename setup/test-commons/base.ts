import type {Config} from '@jest/types';
import path from 'path'; 

// Sync object
const config: Config.InitialOptions = {
  testEnvironment: 'jsdom',
  verbose: true,
  setupFilesAfterEnv: [
    path.resolve(__dirname, 'src', 'setupTests.js'),
  ],
  transform: {
    '\\.[jt]sx?$': ['babel-jest', {
      configFile: path.resolve(__dirname, '.babelrc'),
    }],
  }
};
export default config;
