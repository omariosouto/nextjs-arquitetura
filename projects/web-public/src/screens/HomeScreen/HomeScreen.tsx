import { Text } from "@alura/design-system/components/Text";
import { sum } from "@alura/utils/math/sum";

export function HomeScreen() {
  return (
    <main>
      <Text tag="h1">Home</Text>
      <Text tag="p">
        Importando modulo local: @alura/utils/math/sum sum(2,2): {sum(2, 2)}
      </Text>
    </main>
  );
}
