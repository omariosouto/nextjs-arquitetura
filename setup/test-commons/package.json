{
  "name": "@alura/test-commons",
  "version": "0.0.0",
  "private": true,
  "main": "index.js",
  "files": [
    "base.ts"
  ],
  "devDependencies": {
    "jest-styled-components": "7.0.8",
    "@testing-library/react": "13.2.0",
    "jest": "27.0.0",
    "ts-node": "10.7.0",
    "babel-jest": "28.0.3",
    "@babel/core": "7.17.10",
    "@babel/preset-env": "7.17.10",
    "@babel/preset-react": "7.16.7",
    "@babel/preset-typescript": "7.16.7"
  },
  "scripts": {
    "test": "echo 'Not implemented'",
    "lint": "echo 'Not implemented'"
  }
}
