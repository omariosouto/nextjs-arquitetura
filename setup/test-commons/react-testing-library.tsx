import { render as rtlRender } from '@testing-library/react'

function DefaultWrapper({ children }) {
  return children;
}

export const customRender = (AllTheProviders = DefaultWrapper) => {
  return (ui, options = {}) => {
    return rtlRender(ui, {wrapper: AllTheProviders, ...options});
  }
}

// re-export everything
export * from '@testing-library/react'

// override render method
export const render = () => console.error('Please create your own render method using customRender()');
