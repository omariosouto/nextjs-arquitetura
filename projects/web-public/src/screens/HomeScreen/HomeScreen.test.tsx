import { customRender } from "@alura/test-commons/react-testing-library";
import { HomeScreen } from "./HomeScreen";

function CustomWrapper({ children }) {
  return <section>{children}</section>;
}
const render = customRender(CustomWrapper);

describe("<HomeScreen />", () => {
  it("renders the page", () => {
    const { container } = render(<HomeScreen />);

    expect(container).toMatchSnapshot();
  });
});
