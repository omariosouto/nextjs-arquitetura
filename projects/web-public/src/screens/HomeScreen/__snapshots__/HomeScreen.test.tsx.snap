// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders the page 1`] = `
.c0 {
  font-family: sans-serif;
}

<div>
  <main>
    <h1
      class="c0"
    >
      Home
    </h1>
    <p
      class="c0"
    >
      Importando modulo local: @alura/utils/math/sum sum(2,2): 
      4
    </p>
  </main>
</div>
`;
